### PR TITLE
Finish docs

### LINKED ISSUE #12: Finish writing documentation
The missing pieces are listed in the Readme. Additionally maybe adding a section with real examples might be useful.

### PR DESCRIPTION
**Pull request by Odrakir**
Closes #12 

Jus that. This PR finishes all I had planned for the docs.

Still this document should be updated frequently.

Please read it, make sure you understand it and raise any concerns.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -285,12 +285,103 @@ As you can see the state is mapped from the global one to the local one and back
...
+
+Apple recommends reducing the number of dynamic frameworks our app loads to reducer launch times. In order to have modularity and fast compile times while at the same time keep launch times as small as possible we use a mix of static and dynamic frameworks.
+
+Basically every framework in the app is a static framework (Build Settings -> Mach-0 Type -> Static Library) except for the `TogglTrack` and the `Assets`. These two dynamic frameworks will be embedded in the app, while the rest (the static frameworks) will be linked but not embedded in `TogglTrack`.
+
+Dependencies between frameworks will also need to be linked where they are needed.
+
+External dependencies are currently linked into a dynamic framework that takes care of linking them when needed and it's also embedded into the main app. This will probably change as we move from Swift Package Manager into Cocoapods.
+
+`Assets` has to be dynamic because static frameworks don't allow to those type of resources. Still this makes it difficult to work with playgrounds and we should review this.
```
Comment (sofokli1) [inline README.md]: `don't allow to ...` what? I think something is missing there

Comment (Odrakir) [reply README.md]: Fixed

File: README.md [hunk 2]
```diff
@@ -285,12 +285,103 @@ As you can see the state is mapped from the global one to the local one and back
...
+
+// Somewhere else we "apply" it to the main app's reducer:
+let globalReducer = logging(appReducer)
+```
+
+Analytics is something we will likely develop as a high-order reducer too.
 
 ## Features
 
+Understand feature as a part of the app that needs its own Store view and its own reducer. This limits are blurry, but this usually means every time you present something modally, that's a new feature, but each one of the screens on that modal navigation are all part of the same feature.
```
Comment (dshr) [inline README.md]:
```suggestion
Understand feature as a part of the app that needs its own Store view and its own reducer. The limits are blurry, but this usually means every time you present something modally, that's a new feature, but each one of the screens on that modal navigation are all part of the same feature.
```

File: README.md [hunk 3]
```diff
@@ -285,12 +285,103 @@ As you can see the state is mapped from the global one to the local one and back
...
+        state.route = AppRoute.onboarding
+    }
+    return .empty
+```
+
 # Project Structure
 
 ## Frameworks & Modules
+
+Apple recommends reducing the number of dynamic frameworks our app loads to reducer launch times. In order to have modularity and fast compile times while at the same time keep launch times as small as possible we use a mix of static and dynamic frameworks.
```
Comment (dshr) [inline README.md]:
```suggestion
Apple recommends reducing the number of dynamic frameworks our app loads to reduce launch times. In order to have modularity and fast compile times while at the same time keeping launch times as small as possible we use a mix of static and dynamic frameworks.
```

**Pull request merged**